client/shaders/object_shader/opengl_fragment.glsl

```diff
@@ -343,36 +474,62 @@ void main(void)
 	color = base.rgb;
 	vec4 col = vec4(color.rgb, base.a);
 	col.rgb *= varColor.rgb;
-	col.rgb *= emissiveColor.rgb * vIDiff;
+	col.rgb *= vIDiff;
 
 #ifdef ENABLE_DYNAMIC_SHADOWS
 	float shadow_int = 0.0;
 	vec3 shadow_color = vec3(0.0, 0.0, 0.0);
 	vec3 posLightSpace = getLightSpacePosition();
 
+	float distance_rate = (1 - pow(clamp(2.0 * length(posLightSpace.xy - 0.5),0.0,1.0), 20.0));
+	float f_adj_shadow_strength = max(adj_shadow_strength-mtsmoothstep(0.9,1.1,  posLightSpace.z  ),0.0);
+
+	if (distance_rate > 1e-7) {
+	
 #ifdef COLORED_SHADOWS
-	vec4 visibility;
-	if (cosLight > 0.0)
-		visibility = getShadowColor(ShadowMapSampler, posLightSpace.xy, posLightSpace.z);
-	else
-		visibility = vec4(1.0, 0.0, 0.0, 0.0);
-	shadow_int = visibility.r;
-	shadow_color = visibility.gba;
+		vec4 visibility;
+		if (cosLight > 0.0)
+			visibility = getShadowColor(ShadowMapSampler, posLightSpace.xy, posLightSpace.z);
+		else
+			visibility = vec4(1.0, 0.0, 0.0, 0.0);
+		shadow_int = visibility.r;
+		shadow_color = visibility.gba;
 #else
-	shadow_int = getShadow(ShadowMapSampler, posLightSpace.xy, posLightSpace.z);
+		if (cosLight > 0.0)
+			shadow_int = getShadow(ShadowMapSampler, posLightSpace.xy, posLightSpace.z);
+		else
+			shadow_int = 1.0;
 #endif
+		shadow_int *= distance_rate;
+		shadow_int = clamp(shadow_int, 0.0, 1.0);
 
-	if (f_normal_length != 0 && cosLight <= 0.001) {
-		shadow_int = clamp(shadow_int + 0.5 * abs(cosLight), 0.0, 1.0);
 	}
 
-	shadow_int = 1.0 - (shadow_int * adj_shadow_strength);
+	// turns out that nightRatio falls off much faster than
+	// actual brightness of artificial light in relation to natual light.
+	// Power ratio was measured on torches in MTG (brightness = 14).
+	float adjusted_night_ratio = pow(nightRatio, 0.6);
```

Comment:
Should we do the same as we do in the nodes shader and make sure nightRatio is never negative?

Reply:
Fixed